utils/script-python.c

```diff
@@ -139,13 +139,15 @@ static int set_python_path(char *py_pathname)
 static int import_python_module(char *py_pathname)
 {
 	PyObject *pName;
-	char *py_basename = basename(py_pathname);
+	char *py_basename = xstrdup(basename(py_pathname));
 
 	pr_dbg("%s(%s)\n", __func__, py_pathname);
```

Comment:
Could you please show me what it prints before applying this patch?

Reply:
If I don't add this change, the output shows as below:
```
$ uftrace -S scripts/report-libcall.py -v /usr/bin/gcc ~/hello.c
    ...
mcount: initializing mcount library
script: script_init(scripts/report-libcall.py)
script: script_init_for_python("scripts/report-libcall.py")
script: set_python_path("scripts/report-libcall.py")
script: import_python_module("scripts/report-libcall.py")
script: python initialization finished
mcount: mcount setup done
mcount: initializing mcount library
script: script_init(scripts/report-libcall)
scripts/report-libcall: No such file or directory
mcount: mcount setup done
mcount: shmem_finish: tid: 752 seqnum = 18 curr = 0, nr_buf = 2 max_buf = 2
mcount: destroy plthook special function index
mcount: initializing mcount library
script: script_init(scripts/report-libcall)
scripts/report-libcall: No such file or directory
mcount: mcount setup done
mcount: shmem_finish: tid: 753 seqnum = 3 curr = 1, nr_buf = 2 max_buf = 2
mcount: destroy plthook special function index
mcount: initializing mcount library
script: script_init(scripts/report-libcall)
scripts/report-libcall: No such file or directory
mcount: mcount setup done
mcount: initializing mcount library
script: script_init(scripts/report-libcall)
scripts/report-libcall: No such file or directory
mcount: mcount setup done
mcount: shmem_finish: tid: 756 seqnum = 10 curr = 0, nr_buf = 2 max_buf = 2
mcount: destroy plthook special function index
mcount: shmem_finish: tid: 754 seqnum = 0 curr = 0, nr_buf = 2 max_buf = 2
mcount: destroy plthook special function index
child terminated by signal: 11: Segmentation fault
    ...
```

Reply:
Ah, this was because the script_pathname is an environment variable.  You fix looks fine then.

Reply:
But following debug messages are duplicated so I think only the first one needs level 1.  Please change others to `pr_dbg2` (or simply delete them).

```
script: script_init("scripts/report-libcall.py")
script: script_init_for_python("scripts/report-libcall.py")
script: set_python_path("scripts/report-libcall.py")
script: import_python_module("scripts/report-libcall.py")
script: python initialization finished
```